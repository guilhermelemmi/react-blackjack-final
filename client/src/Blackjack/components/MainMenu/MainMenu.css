
.mainMenu {
  height: calc(100vh - 40px);
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: #fefefe;
  font-family: Arial, Helvetica, sans-serif;
}

.gameRules {
  max-height: 50vh;
  overflow-y: auto;
  margin-bottom: 20px;
  text-align: left;
}