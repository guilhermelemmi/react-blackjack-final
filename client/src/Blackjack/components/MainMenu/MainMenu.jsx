import Button from "../Button";
import Markdown from 'react-markdown'
import "./MainMenu.css";

function MainMenu({onStart, gameRules}) {
  return (
    <div className="mainMenu">
      <h1>Welcome to React Blackjack</h1>
      {gameRules && <div className="gameRules">
        <Markdown>{gameRules}</Markdown>
      </div>}
      <Button type="primary" onClick={onStart}>
        Play
      </Button>
    </div>
  );
}

export default MainMenu;
