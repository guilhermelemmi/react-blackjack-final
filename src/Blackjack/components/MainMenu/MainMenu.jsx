import Button from "../Button";
import "./MainMenu.css";

function MainMenu({onStart}) {
  return (
    <div className="mainMenu">
      <Button type="primary" onClick={onStart}>
        Play
      </Button>
    </div>
  );
}

export default MainMenu;
