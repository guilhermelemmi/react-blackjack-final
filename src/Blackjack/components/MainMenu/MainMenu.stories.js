import MainMenu from "./MainMenu";

export default {
  component: MainMenu,
};

export const Default = {
  render: () => <MainMenu onStart={() => {}} />,
};
