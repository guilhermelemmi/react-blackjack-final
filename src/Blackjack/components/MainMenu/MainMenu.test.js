import { fireEvent, render, screen } from "@testing-library/react";
import MainMenu from "./MainMenu";

const noop = () => {};

describe("MainMenu", () => {
  test("It renders count and label", async () => {
    const handleClick = jest.fn();
    render(<MainMenu onStart={handleClick} />);
    fireEvent.click(screen.getByText(/Play/i));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("It matches snapshot", async () => {
    const view = render(<MainMenu onStart={noop} />);
    expect(view.container).toMatchSnapshot();
  });
});
