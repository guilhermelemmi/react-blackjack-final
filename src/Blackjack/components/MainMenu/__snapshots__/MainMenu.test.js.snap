// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu It matches snapshot 1`] = `
<div>
  <div
    class="mainMenu"
  >
    <button
      class="button primary"
    >
      Play
    </button>
  </div>
</div>
`;
